ecc/fourq/point_test.go

```diff
@@ -121,6 +179,33 @@ func TestScalarMult(t *testing.T) {
 			}
 		}
 	})
+	t.Run("mult-non_curve_point_issue", func(t *testing.T) {
+		for i := 0; i < testTimes; i++ {
+			_, _ = rand.Read(k[:])
+			Q.random()
+			P.ScalarMult(&k, &Q)
+			if !P.IsOnCurve() {
+				t.Fatalf("Point is not on curve: %X\n", P)
+			}
+		}
+	})
+	t.Run("unmarshal-faulty-point", func(t *testing.T) {
+		// This test demonstrates that it is possible to find points which are unmarshalled
+		// successfully, but are not on the curve.
+		var marshalledPoint [Size]byte
+		for i := 0; i < testTimes; i++ {
+			_, _ = rand.Read(marshalledPoint[:])
+			unmarshalledP := Point{}
+			ok := unmarshalledP.Unmarshal(&marshalledPoint)
+			isOnCurve := unmarshalledP.IsOnCurve()
+			switch true {
+			case ok && !isOnCurve:
+				t.Fatalf("unmarshall ok, but not on curve: %v\n", unmarshalledP)
```

Comment:
In commit `fourq: Test showing point unmarshlling fails.`

nit: "unmarshal" or "unmarshalled"
```suggestion
				t.Fatalf("unmarshal ok, but not on curve: %v\n", unmarshalledP)
```

same comment below.